worlds/mmx3/__init__.py

```diff
@@ -255,64 +255,126 @@ def set_rules(self):
                 if "Mega Man X3" in self.multiworld.re_gen_passthrough:
                     slot_data = self.multiworld.re_gen_passthrough["Mega Man X3"]
                     self.boss_weaknesses = slot_data["weakness_rules"]
+                    print (self.boss_weaknesses)
+                    self.boss_weakness_strictness = slot_data["boss_weakness_strictness"]
+                    self.pickupsanity = slot_data["pickupsanity"]
+                    self.jammed_buster = slot_data["jammed_buster"]
+                    self.logic_boss_weakness = slot_data["logic_boss_weakness"]
```

Comment:
I'd recommend putting this stuff in `generate_early` instead -- that way everything after `generate_early` can catch the option changes